### PR TITLE
chore(shared, dal, api, worker, application-generic): remove the unwanted expiredAt in favour of atlas triggers

### PR DESCRIPTION
**Pull request by merrcury**
### What changed? Why was the change needed?

This fixes INF-464

### CODE REVIEW

**Review by SokratisVidros — APPROVED** [review]
@scopsy and @djabarovgeorge PTAL

File: libs/dal/src/repositories/base-repository.ts [hunk 1]
```diff
@@ -222,41 +222,7 @@ export class BaseRepository<T_DBModel, T_MappedEntity, T_Enforcement> {
...
-    if (data.expireAt) {
-      startDate = new Date(data.expireAt);
-    }
-
-    switch (modelName) {
-      case 'Message':
-        if (data.channel === 'in_app') {
-          return addDays(
-            startDate,
-            Number(process.env.MESSAGE_IN_APP_RETENTION_DAYS || DEFAULT_MESSAGE_IN_APP_RETENTION_DAYS)
```
Comment (SokratisVidros) [inline libs/dal/src/repositories/base-repository.ts:236]: Please search for `MESSAGE_IN_APP_RETENTION_DAYS` occurrences in code and delete the env var completely if it's not used elsewhere.

File: libs/dal/src/repositories/base-repository.ts [hunk 2]
```diff
@@ -222,41 +222,7 @@ export class BaseRepository<T_DBModel, T_MappedEntity, T_Enforcement> {
...
-          );
-        } else {
-          return addDays(
-            startDate,
-            Number(process.env.MESSAGE_GENERIC_RETENTION_DAYS || DEFAULT_MESSAGE_GENERIC_RETENTION_DAYS)
```
Comment (SokratisVidros) [inline libs/dal/src/repositories/base-repository.ts:241]: Ditto

File: libs/dal/src/repositories/base-repository.ts (hunk 1 above)
Comment (merrcury) [reply libs/dal/src/repositories/base-repository.ts:236]: Made this change

File: libs/dal/src/repositories/base-repository.ts (hunk 2 above)
Comment (merrcury) [reply libs/dal/src/repositories/base-repository.ts:241]: Made this change

**Review by SokratisVidros — APPROVED** [review]

File: libs/dal/src/repositories/execution-details/execution-details.schema.ts [hunk 3]
```diff
@@ -112,8 +111,6 @@ executionDetailsSchema.index({
   _environmentId: 1,
 });
 
-executionDetailsSchema.index({ expireAt: 1 }, getTTLOptions());
```
Comment (rifont) [inline libs/dal/src/repositories/execution-details/execution-details.schema.ts:115]: Small note, let's make sure we remove this index from the Atlas console after releasing to Prod.

File: libs/dal/src/repositories/job/job.schema.ts [hunk 4]
```diff
@@ -128,15 +128,12 @@ const jobSchema = new Schema<JobDBModel>(
...
       type: Schema.Types.String,
     },
-    expireAt: Schema.Types.Date,
     stepOutput: Schema.Types.Mixed,
     preferences: Schema.Types.Mixed,
   },
   schemaOptions
 );
 
-jobSchema.index({ expireAt: 1 }, getTTLOptions());
```
Comment (rifont) [inline libs/dal/src/repositories/job/job.schema.ts:138]: Same applies here, remove the index after release to Prod.

File: libs/dal/src/repositories/message/message.schema.ts [hunk 5]
```diff
@@ -116,15 +116,12 @@ const messageSchema = new Schema<MessageDBModel>(
...
       ref: 'Subscriber',
     },
-    expireAt: Schema.Types.Date,
     tags: [Schema.Types.String],
     avatar: Schema.Types.String,
   },
   schemaOptions
 );
 
-messageSchema.index({ expireAt: 1 }, getTTLOptions());
```
Comment (rifont) [inline libs/dal/src/repositories/message/message.schema.ts:126]: Ditto, remove after release to Prod

File: libs/dal/src/repositories/notification/notification.schema.ts [hunk 6]
```diff
@@ -50,8 +49,6 @@ const notificationSchema = new Schema<NotificationDBModel>(
   schemaOptions
 );
 
-notificationSchema.index({ expireAt: 1 }, getTTLOptions());
```
Comment (rifont) [inline libs/dal/src/repositories/notification/notification.schema.ts:53]: Ditto, remove after release to Prod

**Review by rifont — APPROVED** [review]

File: libs/dal/src/repositories/execution-details/execution-details.schema.ts (hunk 3 above)
Comment (merrcury) [reply libs/dal/src/repositories/execution-details/execution-details.schema.ts:115]: Captain

File: libs/dal/src/repositories/job/job.schema.ts (hunk 4 above)
Comment (merrcury) [reply libs/dal/src/repositories/job/job.schema.ts:138]: Yep

File: libs/dal/src/repositories/message/message.schema.ts (hunk 5 above)
Comment (merrcury) [reply libs/dal/src/repositories/message/message.schema.ts:126]: Yep

File: libs/dal/src/repositories/notification/notification.schema.ts (hunk 6 above)
Comment (merrcury) [reply libs/dal/src/repositories/notification/notification.schema.ts:53]: Yep

**Pull request merged**